src/interface.jl

```diff
@@ -2,83 +2,101 @@
+    hasdensity(d)::Bool
 
-Returns `true` if `type` is compatible with the `DensityInterface` interface,
-otherwise returns `false` by default.
+Return `true` if `d` is compatible with the `DensityInterface` interface.
+
+`hasdensity(d) == true` implies that `d` is either a density itself or has an
+associated density, e.g. a probability density function or a Radon–Nikodym
+derivative with respect to an implicit base measure. It also implies that the
+value of that density at given points can be calculated via
+[`logdensityof`](@ref) and [`densityof`](@ref).
+```
 """
-function isdensitytype end
-export isdensitytype
+function hasdensity end
+export hasdensity
 
-@inline isdensitytype(::Type) = false
+@inline hasdensity(::Any) = false
 
-function check_if_densitytype(type::Type)
-    isdensitytype(type) || throw("Type $(type) is not compatible with DensityInterface")
+function check_hasdensity(d)
+    hasdensity(d) || throw("Object of type $(typeof(d)) is not compatible with DensityInterface")
 end
 
 
 """
-    logdensityof(density, x)::Real
-    logdensityof(density)
+    logdensityof(d, x)::Real
 
-Computes the logarithmic value of `density` at a given point `x`, resp.
-returns a function that does so:
+Computes the logarithmic value of density `d` or it's associated density
+at a given point `x`.
 
-```julia
-logy = logdensityof(some_density, x)
-logdensityof(some_density, x) == logdensityof(some_density)(x)
-```
+```jldoctest a
+julia> hasdensity(d)
+true
 
-and
-
-```julia
-log_f = logdensityof(density)
-log_f(x) == logdensityof(density, x)
-logfuncdensity(log_f) == density
+julia> logy = logdensityof(d, x); logy isa Real
+true
 ```
 
-`logdensityof(density)` defaults to `Base.Fix1(logdensityof, density)`, but
-may be specialized for a density type. If so, [`logfuncdensity`](@ref) will
-typically have to be specialized for the return type of `logdensityof` as
-well.
-    
-The following identity must always hold:
-
-```julia
-logfuncdensity(logdensityof(density)) == density
-```    
+See also [`hasdensity`](@ref) and [`densityof`](@ref).
 """
 function logdensityof end
 export logdensityof
 
-function logdensityof(density)
-    check_if_densitytype(typeof(density))
-    Base.Fix1(logdensityof, density)
+"""
+    logdensityof(d)
+
+Return a function that computes the logarithmic value of density `d`
+or its associated density at a given point.
+
+```jldoctest a
+julia> log_f = logdensityof(d); log_f isa Function
+true
+
+julia> log_f(x) == logdensityof(d, x)
+true
+```
+
+`logdensityof(d)` defaults to `Base.Fix1(logdensityof, d)`, but may be
+specialized. If so, [`logfuncdensity`](@ref) will typically have to be
+specialized for the return type of `logdensityof` as well.
+
+[`logfuncdensity`](@ref) is the inverse of `logdensityof`, so
+`logfuncdensity(log_f)` must be equivalent to `d`.
+"""
+function logdensityof(d)
+    check_hasdensity(d)
+    Base.Fix1(logdensityof, d)
 end
 
 
 """
     logfuncdensity(log_f)
```

Comment:
This seems like the `∫exp` function in MeasureTheory, which takes a function and a base measure, returning a new measure. I think in order for us to use this, we'd need to be able to extend it similarly

Reply:
Deferring this to future discussions (for now, it'll simply be the "curried" version of `logdensityof`).